src/components/ErrorNotification/ErrorNotification.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { memo, useCallback, useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import { AppContext } from '../../context/AppContext';
+
+export const ErrorNotification: React.FC = memo(() => {
+  const { state, dispatch } = useContext(AppContext);
+
+  const hideErrors = useCallback(() => {
+    dispatch({ type: 'CLEAR_ERRORS' });
+  }, [dispatch]);
+
+  const activeError = state.errors.find(error => error.value);
+
+  useEffect(() => {
+    if (activeError) {
+      const timer = setTimeout(() => {
+        hideErrors();
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+
+    return;
+  }, [activeError, hideErrors]);
```

Comment:
```suggestion
  useEffect(() => {
  let timer: number;
  
    if (activeError) {
       timer = setTimeout(() => {
        hideErrors();
      }, 3000);
    }

    return () => clearTimeout(timer);
  }, [activeError, hideErrors]);
```
this would be more accurate